src/coreclr/debug/createdump/createdumpwindows.cpp

```diff
@@ -17,6 +27,16 @@ CreateDump(const char* dumpPathTemplate, int pid, const char* dumpType, MINIDUMP
     ArrayHolder<char> pszName = new char[MAX_LONGPATH + 1];
     std::string dumpPath;
 
+    // On Windows, createdump is restricted for security reasons to only the .NET process (parent process) that launched createdump
+    PROCESS_BASIC_INFORMATION_ processInformation;
+    NTSTATUS status = NtQueryInformationProcess(GetCurrentProcess(), PROCESSINFOCLASS::ProcessBasicInformation, &processInformation, sizeof(processInformation), NULL);
```

Comment:
Not always great, but can `CreateToolhelp32Snapshot` + Iterating work. `NtQueryInformationProcess` is marked obsolete and internal/unstable.

Reply:
The BCL pinvokes to it here: 
https://github.com/dotnet/runtime/blob/b84a05779aeadf7522e532a970094e5e36fbf5f9/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Win32.cs#L361

It look like it is only used to kill process trees.

I really wanted to avoid CreateToolhelp32Snapshot because of the complexity of having to do the ERROR_PARTIAL_COPY/ERROR_BAD_LENGTH error loop and the process iteration.

Reply:
Can the parent process pass a handle to itself to createdump?